doc/sphinx/language/core/basic.rst

```diff
@@ -0,0 +1,428 @@
+        | @term100
+        term100 ::= @term_cast
+        | @term10
+        term10 ::= @term_application
+        | @one_term
+        one_term ::= @ @qualid_annotated
+        | @term1
+        term1 ::= @term_projection
+        | @term0 % @scope
+        | @term0
+        term0 ::= @qualid_annotated
+        | @sort
+        | @primitive_notations
+        | @term_evar
+        | @term_match
+        | @term_record
+        | @term_generalizing
+        | @term_ltac
+        | ( @term )
+        qualid_annotated ::= @qualid {? @univ_annot }
+
+     .. note::
+
+        Many commands and tactics use :n:`@one_term` rather than
+        :n:`@term`.  The former need to be enclosed in parentheses
+        unless they're very simple, such as a single identifier.  This
+        avoids confusing a space-separated list of terms or
+        identifiers with a :n:`@term_application`.
+
+   type
+
+     Types are a sub-class of the syntactical class of terms:
+
+     .. prodn::
+        type ::= @term
+
+     Not every term is a type, but every term should have an
+     associated type, which is determined by applying a set of typing
+     rules.  If that's indeed the case, we say that the term is
+     :gdef:`well-typed`.
+
+     Intuitively, types may be viewed as sets containing terms.  We
+     say that a type is :gdef:`inhabited` if it contains at least one
+     term (i.e. if we can find a term which is associated to this
+     type).
+
+     Formally, this gives rise to logical foundations which are not
+     the same as the well-known `"set theory"
+     <https://en.wikipedia.org/wiki/Set_theory>`_.  Coq is based on
+     the Calculus of Inductive Construction which is a `"type theory"
+     <https://en.wikipedia.org/wiki/Type_theory>`_.
+
+   sentence
+
+     Coq documents are made of a series of sentences, which are
+     :term:`commands <command>` or :term:`tactics <tactic>`, generally
+     terminated with a period and optionally decorated with
+     :term:`attributes <attribute>`.
+
+     .. insertprodn vernacular sentence
+
+     .. prodn::
+        document ::= {* @sentence }
+        sentence ::= {? @attributes } @command .
+        | {? @attributes } {? @num : } @query_command .
+        | {? @attributes } {? @toplevel_selector } @ltac_expr {| . | ... }
+        | @control_command
+
+   command
+
+     A :production:`command` can be used to modify the state of a Coq
+     document, for instance by declaring a new object, or to get
+     information about the current state.
+
+     By convention, command names begin with uppercase letters.
+     Commands appear in the HTML documentation in blue or gray boxes
+     after the label "Command".  In the pdf, they appear after the
+     boldface label "Command:".  Commands are listed in the
+     :ref:`command_index`.
+
+   tactic
+
+     Tactics specify how to transform the current proof state as a
+     step in creating a proof.  They are syntactically valid only when
```

Comment:
Is it actually a syntax error to use these when not in proof mode?  If so, this gives another example of non-context-freeness in conjunction with the deprecated flag Refine Instance Mode, IIRC

Reply:
Yes. But to me this is an example of non-context-freeness even without things like `Refine Instance Mode` (which has been removed BTW). I mention it already in the first paragraph on syntax:

>Furthermore, some parsing rules are only activated in certain contexts (interactive proof mode, custom entries...).